x/incentive/keeper/abci.go

```diff
@@ -160,14 +238,21 @@ func (k Keeper) IsLPRewardsDistributionEpoch(ctx sdk.Context) (bool, types.Incen
 
 	// Increase current epoch of Stake incentive param
 	lpIncentive.CurrentEpochInBlocks = lpIncentive.CurrentEpochInBlocks.Add(lpIncentive.DistributionEpochInBlocks)
-	if lpIncentive.CurrentEpochInBlocks.GTE(lpIncentive.TotalBlocksPerYear) {
-		if len(params.StakeIncentives) > 1 {
+	if lpIncentive.CurrentEpochInBlocks.GTE(lpIncentive.TotalBlocksPerYear) || curBlockHeight.GT(lpIncentive.TotalBlocksPerYear.Add(lpIncentive.DistributionStartBlock)) {
+		if len(params.LpIncentives) > 1 {
 			params.LpIncentives = params.LpIncentives[1:]
+			k.SetParams(ctx, params)
+			return false, types.IncentiveInfo{}
 		} else {
+			params.LpIncentives = []types.IncentiveInfo(nil)
+			k.SetParams(ctx, params)
 			return false, types.IncentiveInfo{}
 		}
 	}
 
+	params.LpIncentives[0].CurrentEpochInBlocks = lpIncentive.CurrentEpochInBlocks
```

Comment:
Is there any case LpIncentives[1] is used as well? It would be nice to add comments on what is LpIncentives[0] and what are the rest of [1:]

Reply:
LpIncentives[0] - this year
LpIncentives[1] - next year